### PR TITLE
Default PyTorch Hub to `autocast(False)`

### LINKED ISSUE #5924: No detection from example images
Hi I am new to Yolo and I am trying to run the model on my computer, the model does not detect any object from the two example images (i.e. bus and zidane). I used models of different sizes but none of them succeeded. I wonder what went wrong and how to fix it. Here are some details about my environment:

  Windows 10
  Python 3.8
  Torch 1.10.0

Here are what Yolo prints when detecting objects from bus.jpg:

Using cache found in /.cache\torch\hub\ultralytics_yolov5_master
YOLOv5  2021-12-8 torch 1.10.0 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
Fusing layers... 
Model Summary: 367 layers, 46533693 parameters, 0 gradients
Adding AutoShape... 
image 1/1: 1080x810 **(no detections)**
Speed: 17.0ms pre-process, 307.2ms inference, 0.0ms NMS per image at shape (1, 3, 640, 480)

### PR DESCRIPTION
**Pull request by glenn-jocher**
Multiple users are experiencing Windows/Conda/CUDA 11 compatibility issue with PyTorch AMP and FP16 inference causing zero detections. This can be solved by downgrading to CUDA 10, but to ensure new users do not experience issues I am defaulting PyTorch Hub inference to FP32. AMP inference can be enabled using `model.amp = True`.

Usage:
```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, custom
model.amp = False  # <--- disable AMP inference (solves Windows/conda/CUDA11 issues)
model.amp = True  # <--- enable AMP inference

# Images
img = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced inference with optional Automatic Mixed Precision (AMP) in the YOLOv5 AutoShape class.

### 📊 Key Changes
- Added a flag for AMP inference to the AutoShape class.
- Modified the forward method to conditionally enable AMP based on the new flag and device type.

### 🎯 Purpose & Impact
- **Improved Performance**: AMP can speed up inference and reduce memory usage on compatible GPUs, leading to faster and more efficient model predictions.
- **Flexibility**: Users can now choose to enable AMP during inference, providing greater control based on their hardware capabilities and requirements. This can be particularly useful for deploying models in production environments.
- **Broader Compatibility**: By making AMP optional and conditional on the device type, the changes preserve functionality for users running on CPUs or hardware that does not support AMP.

### CODE REVIEW

**Comment by Pro100rus32** [comment]
**_CUDA 11, 3070TI_**

This partially fixes the problem of performance degradation on CUDA 11. But alas, it does not fix it completely.

**model.amp = False**
```
fps(every 1s)
59
57
57
47
57
60
59
60
60
59
59
62
61
57
56
54
54
55
59
```
**model.amp = True**
```
fps(every 1s)
29
30
32
31
29
30
29
37
41
40
40
41
40
46
49
48
48
47
48
47
43
45
47
```
